superset-frontend/src/SqlLab/components/AceEditorWrapper.tsx

```diff
@@ -34,41 +33,44 @@ import {
 
 const langTools = ace.acequire('ace/ext/language_tools');
 
-const propTypes = {
-  actions: PropTypes.object.isRequired,
-  onBlur: PropTypes.func,
-  sql: PropTypes.string.isRequired,
-  schemas: PropTypes.array,
-  tables: PropTypes.array,
-  functionNames: PropTypes.array,
-  extendedTables: PropTypes.array,
-  queryEditor: PropTypes.object.isRequired,
-  height: PropTypes.string,
-  hotkeys: PropTypes.arrayOf(
-    PropTypes.shape({
-      key: PropTypes.string.isRequired,
-      descr: PropTypes.string.isRequired,
-      func: PropTypes.func.isRequired,
-    }),
-  ),
-  onChange: PropTypes.func,
+type HotKey = {
+  key: string;
+  descr: string;
+  name: string;
+  func: () => void;
 };
 
-const defaultProps = {
-  onBlur: () => {},
-  onChange: () => {},
-  schemas: [],
-  tables: [],
-  functionNames: [],
-  extendedTables: [],
-};
+interface Props {
+  actions: {
+    queryEditorSetSelectedText: (edit: any, text: null | string) => void;
+    addTable: (queryEditor: any, value: any, schema: any) => void;
+  };
+  autocomplete: boolean;
```

Comment:
naming nit: `hasAutocomplete` or `autocompleteEnabled` perhaps?

Reply:
I like `autocompleteEnabled` between the two, FWIW :)